lib/active_merchant/billing/gateways/card_connect.rb

```diff
@@ -0,0 +1,283 @@
+        post = {}
+        add_reference(post, authorization)
+        commit('void', post)
+      end
+
+      def verify(credit_card, options = {})
+        authorize(0, credit_card, options)
+      end
+
+      def supports_scrubbing?
+        true
+      end
+
+      def scrub(transcript)
+        transcript
+          .gsub(%r((Authorization: Basic )\w+), '\1[FILTERED]')
+          .gsub(%r(("cvv2\\":\\")\d*(&?)), '\1[FILTERED]')
+          .gsub(%r(("merchid\\":\\")\d*(&?)), '\1[FILTERED]')
+          .gsub(%r((&?"account\\":\\")\d*(&?)), '\1[FILTERED]')
+          .gsub(%r((&?"token\\":\\")\d*(&?)), '\1[FILTERED]')
+      end
+
+      private
+
+      def add_customer_data(post, options)
+        post[:email] = options[:email] if options[:email]
+      end
+
+      def add_address(post, options)
+        address = options[:billing_address] || options[:address]
+        if address
+          post[:address] = address[:address1] if address[:address1]
+          post[:city] = address[:city] if address[:city]
+          post[:region] = address[:state] if address[:state]
+          post[:country] = address[:country] if address[:country]
+          post[:postal] = address[:zip] if address[:zip]
+          post[:phone] = address[:phone] if address[:phone]
+        end
+      end
+
+      def add_money(post, money)
+        post[:amount] = amount(money)
+      end
+
+      def add_currency(post, money, options)
+        post[:currency] = (options[:currency] || currency(money))
+      end
+
+      def add_invoice(post, options)
+        post[:orderid] = options[:order_id]
+        post[:ecomind] = (options[:recurring] ? 'R' : 'E')
+      end
+
+      def add_payment(post, payment)
+        post[:name] = payment.name
+        if card_brand(payment) == 'check'
+          add_echeck(post, payment)
+        else
+          post[:account] = payment.number
+          post[:expiry] = expdate(payment)
+          post[:cvv2] = payment.verification_value
+        end
+      end
+
+      def add_echeck(post, payment)
+        post[:accttype] = 'ECHK'
+        post[:account] = payment.account_number
+        post[:bankaba] = payment.routing_number
+      end
+
+      def add_reference(post, authorization)
+        post[:retref] = authorization
+      end
+
+      def add_additional_data(post, options)
+        post[:ponumber] = options[:po_number]
+        post[:taxamnt] = options[:tax_amount] if options[:tax_amount]
+        post[:frtamnt] = options[:freight_amount] if options[:freight_amount]
+        post[:dutyamnt] = options[:duty_amount] if options[:duty_amount]
+        post[:orderdate] = options[:order_date] if options[:order_date]
+        post[:shipfromzip] = options[:ship_from_zip] if options[:ship_from_zip]
+        if (shipping_address = options[:shipping_address])
+          post[:shiptozip] = shipping_address[:zip]
+          post[:shiptocountry] = shipping_address[:country]
+        end
+        if options[:items]
+          post[:items] = []
```

Comment:
This works fine, but I had to read it twice to grok what you're actually doing is a mapping operation; might be worth converting it to

    post[:items] = options[:items].map do |item| ...

to make it clearer that's what's happening.